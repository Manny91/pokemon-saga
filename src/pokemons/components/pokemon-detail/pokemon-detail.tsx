import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import { PokemonDetailContainerProps } from "./pokemons-detail.container";
import { StyledLink } from "../../pokemons";
import { PokemonListItem } from "../pokemon-list-item/pokemon-list-item";
import styled from "../../../styled-components";
import { PokemonMoveDisplayer } from "../move-displayer/moves-displayer";
import { PokemonMove } from "../../../services/pokemon.service";
import { PokemonImage } from "../pokemon-image/pokemon-image";
import { PokemonTypeDisplayer } from "../type-displayer/type-displayer";

export const PokemonDetail = ({
  getPokemonDetail,
  pokemon,
  loading,
  selectPokemon,
  getPokemonMove,
  selectMove,
  selectedMove,
  loadingMoves
}: PokemonDetailContainerProps) => {
  const { id } = useParams();
  const evolutions = pokemon?.species.evolutions;
  const [selectedMoveIndex, setSelectedMoveIndex] = useState(0);
  const pokemonImageId = id + "";
  const types = pokemon?.types;
  useEffect(() => {
    if (id && !loading) {
      selectPokemon(id);
      getPokemonDetail(id);
    }
  }, [id]);

  useEffect(() => {
    if (pokemon) {
      const moveName = pokemon.moves[0].move.name;
      selectMove(moveName);
      getPokemonMove(moveName);
    }
  }, [pokemon]);

  useEffect(() => {
    if (pokemon) {
      const pokemonMoves = pokemon.moves;
      const move: PokemonMove = pokemonMoves[selectedMoveIndex];
      const pokeMove = pokemonMoves.find(
        (pokeMove: PokemonMove) => pokeMove.move.name === move.move.name
      );
      if (pokeMove) {
        selectMove(pokeMove.move.name);
        getPokemonMove(pokeMove.move.name);
      }
    }
  }, [selectedMoveIndex]);

  const handleMoveNext = () => {
    setSelectedMoveIndex(selectedMoveIndex + 1);
  };
  const handleMovePrevious = () => {
    setSelectedMoveIndex(Math.max(0, selectedMoveIndex - 1));
  };
  return (
    <>
      {pokemon && (
        <>
          <DisplayerWrapper>
            <InfoDisplayer>
              <TextDisplayer>
                <TextParagraph> {pokemon.name}</TextParagraph>
                <TextParagraph>{`${pokemon?.height /
                  10} meters`}</TextParagraph>
                <TextParagraph>{`${pokemon?.weight} Kg`}</TextParagraph>
              </TextDisplayer>
              <PokemonImage id={pokemonImageId} />
            </InfoDisplayer>
            <PokemonEvolutionList>
              {evolutions &&
                evolutions.map(({ name, id }) => {
                  return (
                    <StyledLink key={id} to={`/pokemon/${id}/`}>
                      {id && (
                        <PokemonListItem name={name} id={id} showId={false} />
                      )}
                    </StyledLink>
                  );
                })}
            </PokemonEvolutionList>
          </DisplayerWrapper>
          <MoveAndTypeSection>
            <MoveSectionDisplayer>
              <PokemonMoveDisplayer
                handleMoveNext={handleMoveNext}
                handleMovePrevious={handleMovePrevious}
                move={selectedMove}
                loadingMoves={loadingMoves}
              />
            </MoveSectionDisplayer>
            <TypeSectionDisplayer>
              {types && <PokemonTypeDisplayer types={types} />}
            </TypeSectionDisplayer>
          </MoveAndTypeSection>
        </>
      )}
    </>
  );
};
const PokemonEvolutionList = styled.div`
  display: flex;
  flex-direction: column;
  margin: 0px;
  padding: 0px;
  margin: ${props => props.theme.spacing.xs};

  @media ${props => props.theme.media.lg} {
    flex-direction: row;
  }
`;
const DisplayerWrapper = styled.div`
  padding: ${props => props.theme.spacing.xs};
  display: flex;
  flex-direction: column;
  border: groove ${props => props.theme.colors.greenDarkBorder} 3px;
  @media ${props => props.theme.media.md} {
    margin: ${props => props.theme.spacing.sm};
    flex-direction: row;
  }
`;
const MoveAndTypeSection = styled(DisplayerWrapper)``;
const MoveSectionDisplayer = styled(DisplayerWrapper)`
  height: 100px;
  flex-direction: row;
  width: 100%;
  border: none;
  @media ${props => props.theme.media.md} {
    width: 50%;
    padding: 0px;
    margin: 0px;
  }
`;
const TypeSectionDisplayer = styled(MoveSectionDisplayer)`
  padding: 0px;
`;

const TextDisplayer = styled.div`
  display: flex;
  flex-direction: column;
  width: 45%;
`;
const InfoDisplayer = styled.div`
  padding: ${props => props.theme.spacing.sm};
  border: inset ${props => props.theme.colors.greenBorder} 3px;
  display: flex;
  background: ${props => props.theme.colors.greenScreenBackground};
  @media ${props => props.theme.media.md} {
    width: 100%;
  }
  margin: ${props => props.theme.spacing.xs};
`;

const TextParagraph = styled.h3`
  text-transform: capitalize;
  margin: ${props => props.theme.spacing.xs};
`;
