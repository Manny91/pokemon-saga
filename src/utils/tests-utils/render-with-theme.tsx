import React from "react";
import { ThemeProvider } from "../../styled-components";
import { theme } from "../../theme";
import { render } from "@testing-library/react";

export const renderWithTheme = (ui: JSX.Element) => {
  return render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);
};
