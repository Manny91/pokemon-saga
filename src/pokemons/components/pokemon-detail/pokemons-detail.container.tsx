import { connect } from "react-redux";
import { PokemonDetail } from "./pokemon-detail";
import { AppState, PokemonsDispatch } from "../../../store";
import {
  performGetPokemonDetailAction,
  selectPokemonById
} from "../../store/pokemons.actions";
import { PokemonDetail as Pokemon } from "../../../services/pokemon.service";
import { getPokemonsLoading, getPokemonSelected } from "../../store/reducer";

interface DispatchProps {
  getPokemonDetail(pokemonId: string): void;
  selectPokemon(pokemonId: string): void;
}

interface MapStateToProps {
  pokemon?: Pokemon;
  loading: boolean;
}

export type PokemonDetailContainerProps = DispatchProps & MapStateToProps;

function mapStateToProps(state: AppState): MapStateToProps {
  return {
    pokemon: getPokemonSelected(state),
    loading: getPokemonsLoading(state)
  };
}

const mapDispatchToProps = (dispatch: PokemonsDispatch): DispatchProps => ({
  getPokemonDetail: (pokemonId: string) =>
    dispatch(performGetPokemonDetailAction(pokemonId)),
  selectPokemon: (pokemonId: string) => {
    dispatch(selectPokemonById(pokemonId));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(PokemonDetail);
