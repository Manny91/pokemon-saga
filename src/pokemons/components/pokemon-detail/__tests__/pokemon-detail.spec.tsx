import React from "react";
import { PokemonDetailView } from "../pokemon-detail";
import {
  PokemonDetail as Pokemon,
  PokemonSpecieDetail
} from "../../../../services/pokemon.service";
import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";
import { renderWithTheme } from "../../../../utils/tests-utils/render-with-theme";
import { MemoryRouter } from "react-router-dom";

describe("PokemonDetail Component", () => {
  const pokemonSpecie: PokemonSpecieDetail = {
    flavor_text_entries: [
      {
        flavor_text: "test",
        language: { name: "en" }
      }
    ],
    evolution_chain: { url: "evolution-chain-url" },
    evolutions: [
      {
        name: "ivisaur",
        url: "url-test",
        id: "1"
      }
    ],
    id: 1,
    name: "bulbasaur",
    url: "test-url"
  };
  const selectedMove = {
    id: 13,
    name: "razor-wind",
    pp: 10,
    accuracy: 10,
    power: 80,
    url: "https://pokeapi.co/api/v2/move/13/",
    move: { name: "razor-wind", url: "https://pokeapi.co/api/v2/move/13/" }
  };
  const pokemon: Pokemon = {
    abilities: [{ ability: { name: "chlorophyll" } }],
    height: 7,
    id: 1,
    moves: [selectedMove],
    name: "bulbasaur",
    order: 1,
    species: pokemonSpecie,
    sprites: {
      back_default:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/1.png",
      front_default:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/front/1.png",
      back_shiny:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/1.png",
      front_shiny:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/front/1.png"
    },
    stats: [{ base_stat: 45, effort: 0, stat: { name: "speed" } }],
    types: [{ slot: 2, type: { name: "poison" } }],
    weight: 69
  };
  //   const testCases = [];
  it("should render properly", () => {
    const testProps = {
      pokemon: pokemon,
      loading: true,
      selectedMove: selectedMove,
      loadingMoves: false,
      getPokemonDetail: jest.fn(),
      selectPokemon: jest.fn(),
      getPokemonMove: jest.fn(),
      selectMove: jest.fn()
    };
    jest.mock("react-router", () => ({
      useParams: jest.fn().mockReturnValue({ id: "123" })
    }));

    const subject = renderWithTheme(
      <MemoryRouter>
        <PokemonDetailView {...testProps} />
      </MemoryRouter>
    );
    const tree = subject.container;
    expect(tree).toMatchSnapshot();
  });
});
