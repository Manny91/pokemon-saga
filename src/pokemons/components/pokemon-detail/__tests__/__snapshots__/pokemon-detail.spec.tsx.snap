// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetail Component should render properly 1`] = `
.c4 {
  image-rendering: pixelated;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: 2px solid black;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #A5CD53;
  border-width: 3px;
  border-style: inset;
  border-color: #9aa28b;
  font-weight: 500;
  color: black;
  font-family: "VT323";
}

.c7.selected {
  font-weight: bold;
  background-color: #CBEE84;
}

.c7:hover {
  font-weight: bold;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8.reduced {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c9 {
  font-size: 20px;
  text-align: center;
  text-transform: capitalize;
  margin-left: 10px;
}

.c30 {
  width: 4px;
  height: 4px;
  background-color: black;
  border-radius: 1px;
  margin: 0px 2px;
  list-style-type: none;
  -webkit-animation: fZpxSO 1s infinite ease-in-out both;
  -webkit-animation: fZpxSO 1s infinite ease-in-out both;
  animation: fZpxSO 1s infinite ease-in-out both;
}

.c28 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0px;
}

.c28 .c29:nth-child(1) {
  -webkit-animation-delay: -0.35s;
  animation-delay: -0.35s;
}

.c28 .c29:nth-child(2) {
  -webkit-animation-delay: -0.2s;
  animation-delay: -0.2s;
}

.c14 {
  background-color: #A5CD53;
  padding: 10px 20px;
  border-radius: 3px;
  border: inset #9aa28b 3px;
  width: calc(100% - 100px);
}

.c16 {
  text-transform: capitalize;
  margin: 5px 0px;
}

.c17 {
  margin: 5px;
  border-top: 2px solid black;
  list-style-type: none;
  margin: 0px;
  padding: 0;
}

.c18 {
  font-weight: bold;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c20 {
  width: 40px;
  height: 40px;
  border-radius: 50%;
  border: groove grey 3px;
  margin: 5px;
}

.c22 {
  background: linear-gradient(14deg,#afafaf 50%,#ffffff 80%,#afafaf 90%);
  border: groove #4e4e4e 3px;
  width: 100%;
}

.c23 {
  height: 35px;
  text-transform: uppercase;
  text-align: center;
  border: groove #757575 3px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 19px;
  font-weight: bold;
  border-top-right-radius: 3px;
  border-top-left-radius: 3px;
  border-bottom-right-radius: 3px;
  border-bottom-left-radius: 3px;
  margin: 3px;
}

.c23.normal {
  background: #bfbfbf;
}

.c23.fighting {
  background: #d87c58;
}

.c23.flying {
  background: #999ade;
}

.c23.poison {
  background: #925192;
}

.c23.ground {
  background: #dea761;
}

.c23.rock {
  background: #897864;
}

.c23.bug {
  background: #b1c967;
}

.c23.ghost {
  background: #c195dc;
}

.c23.steel {
  background: #49769c;
}

.c23.fire {
  background: #cf1414;
}

.c23.water {
  background: #1689de;
}

.c23.grass {
  background: #47a047;
}

.c23.electric {
  background: #e6b700;
}

.c23.psychic {
  background: #fa43b8;
}

.c23.ice {
  background: #98c3de;
}

.c23.dragon {
  background: #89315d;
}

.c23.dark {
  background: #282433;
}

.c23.fairy {
  background: #dca0ce;
}

.c23.unknown {
  background: #545454;
}

.c23.shadow {
  background: #364163;
}

.c24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  background-color: #A5CD53;
  padding: 15px 20px;
  border-radius: 3px;
  border: inset #9aa28b 3px;
}

.c25 {
  margin: 5px 0px;
  font-weight: bold;
  text-transform: capitalize;
}

.c26 {
  border: 1px solid;
  background-color: #A5CD53;
  padding: 5px;
  border-radius: 3px;
  border: inset #9aa28b 3px;
}

.c26 .c27 {
  margin: 7px;
}

.c31 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #A5CD53;
  padding: 15px 20px;
  border-radius: 3px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 120px;
  border: inset #9aa28b 3px;
}

.c33 {
  margin: 5px 0px;
  font-weight: bold;
  text-transform: capitalize;
}

.c32 {
  text-transform: capitalize;
}

.c6 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0px;
  padding: 0px;
  margin: 5px;
}

.c0 {
  padding: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: groove #4e4e4e 3px;
}

.c10 {
  padding: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: groove #4e4e4e 3px;
}

.c11 {
  padding: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: groove #4e4e4e 3px;
  height: 100px;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  border: none;
}

.c21 {
  padding: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: groove #4e4e4e 3px;
  height: 100px;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  border: none;
  padding: 0px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 45%;
}

.c1 {
  padding: 10px;
  border: inset #9aa28b 3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: linear-gradient( 14deg, rgb(165,205,83) 60%, rgb(193,217,144) 65% );
  margin: 5px;
}

.c3 {
  text-transform: capitalize;
  margin: 5px;
}

.c12 {
  padding: 10px;
  border: inset #9aa28b 3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: linear-gradient( 14deg, rgb(165,205,83) 60%, rgb(193,217,144) 65% );
  margin: 5px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: auto;
  width: 100%;
}

.c13 {
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px;
  cursor: pointer;
}

.c13.selected {
  display: list-item;
}

@media (min-width:992px) {
  .c8.reduced {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    width: 140px;
  }
}

@media (min-width:992px) {
  .c5 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:768px) {
  .c0 {
    margin: 10px;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:768px) {
  .c10 {
    margin: 10px;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:768px) {
  .c11 {
    margin: 10px;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:768px) {
  .c11 {
    width: 50%;
    padding: 0px;
    margin: 0px;
  }
}

@media (min-width:768px) {
  .c21 {
    margin: 10px;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:768px) {
  .c21 {
    width: 50%;
    padding: 0px;
    margin: 0px;
  }
}

@media (min-width:768px) {
  .c1 {
    width: 100%;
  }
}

@media (min-width:768px) {
  .c12 {
    width: 100%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h3
          class="c3"
        >
          Name: 
          bulbasaur
        </h3>
        <h3
          class="c3"
        >
           Order: 
          1
        </h3>
        <h3
          class="c3"
        >
          Height: 
          0.7 meters
        </h3>
        <h3
          class="c3"
        >
          Weight: 
          69 Kg
        </h3>
      </div>
      <img
        class="c4"
        src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/undefined.png"
      />
    </div>
    <div
      class="c5"
    >
      <a
        class="c6"
        href="/pokemon/1/"
      >
        <li
          class="c7"
        >
          <div
            class="c8 reduced"
          >
            <img
              class="c4"
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
            />
            <p
              class="c9"
            >
              ivisaur
            </p>
          </div>
        </li>
      </a>
    </div>
  </div>
  <div
    class="c10"
  >
    <div
      class="c11"
    >
      <div
        class="c12"
      >
        <li
          class="c13 selected"
        >
          razor-wind
        </li>
      </div>
    </div>
    <div
      class="c11"
    >
      <section
        class="c14"
      >
        <div
          class="c15"
        >
          <h4
            class="c16"
          >
            razor-wind
          </h4>
          <ul
            class="c17"
          >
            <li
              class="c18"
            >
              Power : 
              80
            </li>
            <li
              class="c18"
            >
              Accuracy : 
              10
            </li>
            <li
              class="c18"
            >
              PP : 
              10
            </li>
          </ul>
        </div>
        <div
          class=""
        />
      </section>
      <div
        class="c19"
      >
        <button
          class="c20"
        >
          <i
            class="fas fa-caret-up"
          />
        </button>
        <button
          class="c20"
        >
          <i
            class="fas fa-caret-down"
          />
        </button>
      </div>
    </div>
    <div
      class="c21"
    >
      <div
        class="c22"
      >
        <div
          class="sc-fzokOt c23 poison"
        >
          poison
        </div>
      </div>
    </div>
  </div>
  <div
    class="c0"
  >
    <div
      class="c24"
    >
      <p
        class="c25"
      >
        speed.............45
      </p>
    </div>
    <div
      class="c26"
    >
      <ul
        class="c27 c28"
      >
        <li
          class="c29 c30"
        />
        <li
          class="c29 c30"
        />
        <li
          class="c29 c30"
        />
      </ul>
    </div>
    <div
      class="c31"
    >
      <h4
        class="c32"
      >
         Abilities: 
      </h4>
      <p
        class="c33"
      >
        chlorophyll
      </p>
    </div>
  </div>
</div>
`;
